### PR TITLE
Add debugging info for Bug 1646252.

### PR DESCRIPTION
**Pull request by agi**
---

### Pull Request checklist

- [X] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt [hunk 1]
```diff
@@ -156,7 +156,17 @@ class GeckoEngineView @JvmOverloads constructor(
...
-        currentSession?.let { currentGeckoView.setSession(it.geckoSession) }
+        try {
+            currentSession?.let { currentGeckoView.setSession(it.geckoSession) }
+        } catch (e: IllegalStateException) {
+            // This is to debug "Display not attached" crashes
+            val otherActivityClassName =
+                currentSession?.geckoSession?.accessibility?.view?.context?.javaClass?.simpleName
+            val activityClassName = context.javaClass.simpleName
+            val msg =
+                "SET SESSION: Current activity: $activityClassName Other activity: $otherActivityClassName"
```
Comment (csadilek) [inline components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt:167]: We just recently added the hashcodes when logging the activities. I think this is helpful here too so we can tell if it's the same instance or not: https://github.com/mozilla-mobile/android-components/blob/master/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt#L46

Comment (agi) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt]: Done! Thanks.

Comment (csadilek) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt]: Thanks!

**Pull request merged**